src/elements/frame-router.ts

```diff
@@ -103,6 +129,33 @@ class FrameRouterElement extends HTMLElement {
       this.changeRoute(newValue);
     }
   }
+
+  private _handleClientMessage(message: ClientToHost): void {
+    switch (message.msgType) {
+      case 'publish':
+        this._subscriptionManager.dispatchMessage(message.msg);
+        break;
+      case 'lifecycle':
+        this._handleLifecycleMessage(message.msg as LifecycleStage);
+        break;
+      default:
+        this._dispatchClientMessag(message);
+    }
+  }
+
+  private _handleLifecycleMessage(message: LifecycleStage) {
+    if (Lifecycle.isStartedStage(message)) {
```

Comment:
Another different way to check which message was started.  I liked having the stage name just in one spot, but I am open for suggestions.